### ISSUE TITLE
TZInfo with ActiveSupport - Samoa Timezone error

### ISSUE THREAD

**Issue by pedrofurtado** [tzinfo/tzinfo#66]
I am having the same issue as this user, that created this issue in repository of Rails: (https://github.com/rails/rails/issues/27419)

**Steps to reproduce**

In rails console.

`DateTime.now.in_time_zone("Samoa")` Returns a date with a UTC offset of + 14:00
`ActiveSupport::TimeZone.new('Samoa').utc_offset/60/60` Returns -11

**Expected behavior**

`ActiveSupport::TimeZone.new('Samoa').utc_offset/60/60` Should return 14 or 13 based on daylight savings time

**Actual behavior**

`ActiveSupport::TimeZone.new('Samoa').utc_offset/60/60` Returns -11

**System configuration**

Rails version: 5.0.2

Ruby version: 2.3.3

**Comment by philr** [comment]
It looks like there is a bug with the TZInfo zoneinfo data source that is causing the Pacific/Apia time zone (that Active Support calls Samoa) to be represented with a base offset of -11 hours and a daylight savings offset of either +24 or +25 hours (instead of a base offset of +13 hours and a daylight savings offset of either 0 or +1 hour):

```ruby
require 'tzinfo'
TZInfo::DataSource.set(:zoneinfo)
period = TZInfo::Timezone.get('Pacific/Apia').period_for_utc(Time.utc(2017, 3, 6))
period.utc_offset / 3600       # => -11
period.std_offset / 3600       # =>  25 
period.utc_total_offset / 3600 # =>  14
```

The reason for the difference between `DateTime.now.in_time_zone("Samoa")` and `ActiveSupport::TimeZone.new('Samoa').utc_offset/60/60` is that the former is showing the offset including daylight savings time (`utc_total_offset`) and the latter is showing just the base offset (`utc_offset`).

The expected behaviour is that `ActiveSupport::TimeZone.new('Samoa').utc_offset/60/60` would always return 13 regardless of whether daylight savings time is being observed.

Pacific/Apia switched from one side of the International Date Line to the other in December 2011. I suspect that the code that attempts to derive the daylight savings offset from zoneinfo files is not handling this change correctly. 

I'll look into fixing the bug. In the mean time, you can work around the problem by using the `tzinfo-data` gem as TZInfo's data source instead of the system zoneinfo files.

```ruby
require 'tzinfo'
TZInfo::DataSource.set(:ruby)
period = TZInfo::Timezone.get('Pacific/Apia').period_for_utc(Time.utc(2017, 3, 6))
period.utc_offset / 3600       # => 13
period.std_offset / 3600       # =>  1 
period.utc_total_offset / 3600 # => 14
```

To change the data source, you'll need to add a dependency on `tzinfo-data` to your `Gemfile`. By default, Rails ships with this dependency, but limits it to Windows and JRuby. You'll either need to remove the `platforms:` option from the existing `gem 'tzinfo-data'` line if it is there, or add `gem 'tzinfo-data'` to the `Gemfile`.

**Comment by pedrofurtado** [comment]
Thanks @philr !

**Comment by philr** [comment]
The fix for this issue has now been released in [version 1.2.3](https://github.com/tzinfo/tzinfo/releases/tag/v1.2.3).
